### PR TITLE
Refactored Code & Added Database Connection for Login Service

### PR DESCRIPTION
**Pull request by ayanourr**
Got the Dockerfile and Docker Compose setup from my teammate Fatma through email.
Connected the app to a PostgreSQL database and set up the environment variables to make it work.
Restructured the code for better organization.
I also added the connection and service for the login endpoint!

### CODE REVIEW

**Review by FatmaQunnies24 — APPROVED** [review]

File: .gitignore
Comment (FaresAbuIram) [inline .gitignore]: Why did you add this file here?

File: app/main.py
Comment (FaresAbuIram) [inline app/main.py:7]: This code is used for both starting up and shutting down. Why do we need to execute this code during shutdown?

check this https://fastapi.tiangolo.com/tutorial/sql-databases/#create-database-tables-on-startup

File: app/schemas/order_schema.py
Comment (FaresAbuIram) [inline app/schemas/order_schema.py:1]: I believe it's best to create the file when it's needed.

File: app/schemas/product_schema.py
Comment (FaresAbuIram) [inline app/schemas/product_schema.py:1]: I believe it's best to create the file when it's needed.

File: docker-compose.yml
Comment (FaresAbuIram) [inline docker-compose.yml]: add those variables to the environment variables.

File: docker-compose.yml
Comment (FaresAbuIram) [inline docker-compose.yml]: add those variables to the environment variables.

File: app/api/routes/user.py
Comment (FaresAbuIram) [inline app/api/routes/user.py:1]: why??

**Review by FaresAbuIram — CHANGES_REQUESTED** [review]
For future, please keep the PRs shorter. For example, one for adding the database files, one for user changes, and so on.

**Review by sajalahalih — APPROVED** [review]

File: .gitignore
Comment (ayanourr) [reply .gitignore]: I didn’t realize before that I could store database variables in the .env for better security. I’m going to move them there, and then update the .gitignore

File: app/main.py
Comment (FaresAbuIram) [reply app/main.py:7]: I want you to focus on the example in the link. Your approach is good but we have a built-in functions to handle the database startup and shutdown.

I will not say to remove this part but please focus on it

One more update, please add try except and close the database if finally statement

File: app/api/routes/user.py
Comment (FaresAbuIram) [inline app/api/routes/user.py:1]: I think you need to refactor this file as well
I recommend you to keep it in another pr since this one is big enough

**Review by FaresAbuIram — COMMENTED** [review]
I added another two comments

File: app/main.py
Comment (ayanourr) [reply app/main.py:7]: As I read [here](https://fastapi.tiangolo.com/advanced/events/#lifespan), both approaches—using @app.on_event("startup") and the lifespan function—are valid. However, the main difference is that @app.on_event("startup") is synchronous by default, though it can also be made asynchronous if needed. In contrast, the lifespan function is always asynchronous.

File: app/api/routes/user.py
Comment (ayanourr) [reply app/api/routes/user.py:1]: Yes, I was planning to refactor it after getting approval for this PR, which is why I have commented it.

File: app/main.py
Comment (FaresAbuIram) [reply app/main.py:7]: Yes, my point is that it will be better to use the built-in function provided by our framework team

**Review by FaresAbuIram — APPROVED** [review]

**Pull request merged**
